File: benchmark.py
SIZE = 50000

#initialize arrays A, B and C
A = [0] * SIZE
B = [0] * SIZE
C = [0] * SIZE

for i in range(SIZE): #input values into A, B and C
    A[i] = i
    B[i] = SIZE-i
    C[i] = 0

#for each element in C, compute sum of B[j] from 0 to A[i]
for i in range(SIZE):
    total = 0
    for j in range(A[i] + 1):
        total += B[j]
    C[i] = total

checksum = sum(C) #final sum

#output checksum and compare against expected
expected_checksum = 41667916675000
print(f"Computed checksum: {checksum}")

if checksum != expected_checksum:
    print(f"Error: checksum is incorrect. Expected {expected_checksum}, got {checksum}", flush=True)
    import sys
    sys.exit(1) #exit with nonzero error value of 1 if incorrect

File: Lab01/benchmark.c
#include <stdio.h>
#include<stdlib.h>

#define SIZE 50000 //define variables
int A[SIZE];
int B[SIZE];
long long C[SIZE];

int main()
{
	long long checksum = 0; //initialize final sum
	const long long expected_checksum = 41667916675000; //expected value for final sum

	for (int i = 0; i < SIZE; i++) { //initialize A,B and C arrays
		A[i] = i;
		B[i] = SIZE-i;
		C[i] = 0;
	}

	for (int i = 0; i < SIZE; i++) { //for each element in C
		long long sum = 0; //new sum for each C
		for (int j = 0; j <= A[i]; j++) { //sum B from 0 to A[i]
			sum += B[j];
		}
		C[i] = sum; //C[i] is the sum of B from 0 to A[i]
	}

	for (int i = 0; i < SIZE; i++) { //compute final sum of all C components
		checksum += C[i];
	}

	printf("Computed checksum: %lld\n", checksum); //print final sum

	if(checksum != expected_checksum) { //compare final sum to expected sum, exit with return value of 1 if wrong
		fprintf(stderr, "Error: checksum is incorrect. Expected %lld, got %lld\n", expected_checksum, checksum);
		return 1;
	}

	return 0; //return 0 if no error
}
